### ISSUE TITLE
False positive: superfluous-parens

### ISSUE THREAD

**Issue by socketpair** [PyCQA/pylint#5528]
### Bug description
```python
config = {res[0]: res[1].rstrip() for line in entry.read_text().splitlines() if (res := line.split(None, 1))}
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
a.py:41:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
```

### Expected behavior

no warning here

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.9.9 (main, Nov 19 2021, 00:00:00) 
[GCC 11.2.1 20210728 (Red Hat 11.2.1-1)]
```

### OS / Environment

Fedora 34

**Comment by DanielNoord** [comment]
Thanks for the report!

A smaller reproducing code snippet is:
```python
iterable = [1, 2, 3]
my_set = {j for i in iterable if (j := i)}
```

That should help fix and test this more easily 😄

**Comment by Pierre-Sassoulas** [comment]
I think this is fixed in latest versions could you update your pylint version to 2.12.2 ?

**Comment by DanielNoord** [comment]
I can confirm this has been fixed on the latest `main` commit. I used an old version to write the previous example.
